File: 프로그래머스/2/12980. 점프와 순간 이동/점프와 순간 이동.js
function getLeast(n){
    if (n === 0) return 0;
    if (n === 1) return 1;
    if(n % 2 === 0) return getLeast(n / 2)
    if (n % 2 === 1) return getLeast((n - 1) / 2) + 1
}

function solution(n){
    return getLeast(n);
}

File: 프로그래머스/2/12980. 점프와 순간 이동/점프와 순간 이동.py
import sys

sys.setrecursionlimit(10**6)  # 재귀 한도를 크게 설정

dp = {1: 1, 2: 1, 3: 2, 4: 1, 5: 2, 6: 2}

def recursion(k, n):
    if n in dp:
        return dp[n]
    
    if n % 2 == 0:
        return recursion(k, n // 2)
    
    dp[n] = recursion(k, n // 2) + 1
    
    return dp[n]

def solution(n):
    return recursion(0, n)